src/lib/UtapiClient.js

```diff
@@ -113,13 +120,55 @@ export default class UtapiClient {
     }
 
     /**
+     * Utility function to log the metric being pushed.
+     * @param {object} params - params object with metric data
+     * @param {string} method - the name of the method being logged
+     * @param {number} timestamp - normalized timestamp of current time
+     * @param {object} log - Werelogs request logger
+     * @return {undefined}
+     */
+    _logMetric(params, method, timestamp, log) {
+        const logObject = {
+            method: `UtapiClient.${method}`,
+            timestamp,
+        };
+        const metricType = metricTypes.find(type => type in params);
+        logObject[metricType] = params[metricType];
+        log.trace('pushing metric', logObject);
+    }
+
+    /**
+     * Creates an array of parameter objects for each metric type. The number
+     * of objects in the array will be the number of metric types included in
+     * the `params` object.
+     * @param {object} params - params object with metric data
+     * @return {object []} arr - array of parameter objects for push metric call
+     */
+    _getParamsArr(params) {
+        const arr = [];
+        const types = metricTypes.filter(type => type in params);
+        types.forEach(type => {
```

Comment:
looks like you can use a .map here since you are creating a new array with items that correspond to items of the types array